addons/project/static/tests/burndown_chart_tests.js

```diff
@@ -0,0 +1,169 @@
+                    },
+                    records: [
+                        { id: 1, name: 'Todo' },
+                        { id: 2, name: 'In Progress' },
+                        { id: 3, name: 'Done' },
+                    ]
+                }
+            };
+
+            this.burndown_chart = {
+                View: BurndownChartView,
+                model: 'burndown_chart',
+                data: this.data,
+                arch: `<graph type="line">
+                    <field name="date" string="Date" interval="month" type="row"/>
+                    <field name="stage_id"/>
+                    <field name="nb_tasks" type="measure"/>
+                </graph>`,
+            };
+        }
+    }, function () {
+        QUnit.module('BurndownChart');
+
+        QUnit.test('check if stacked in line chart', async function (assert) {
+            assert.expect(3);
+
+            const burndown_chart = await createView(this.burndown_chart);
+
+            assert.strictEqual(burndown_chart.renderer.props.mode, "line", "should be in line chart mode by default");
+            assert.ok(burndown_chart.renderer.props.stacked, "should be stacked by default");
+            assert.ok(burndown_chart.renderer.componentRef.comp._getScaleOptions().yAxes.every(y => y.stacked), 'the stacked property in y axes should be true when the stacked is enabled in line chart');
+
+            burndown_chart.destroy();
+        });
+
+        QUnit.test('check if the stacked button is visible in the line chart', async function (assert) {
+            assert.expect(5);
+
+            const burndown_chart = await createView(this.burndown_chart);
+
+            assert.ok(burndown_chart.renderer.props.stacked, "should be stacked by default");
+            assert.strictEqual(burndown_chart.renderer.props.mode, "line", "should be in line chart mode by default");
+
+            assert.containsN(burndown_chart, '.o_graph_button[data-mode="stack"]', 1, "should find the stacked button in the controller of the view");
+            const $stackButton = burndown_chart.$buttons.find('.o_graph_button[data-mode="stack"]');
+            assert.doesNotHaveClass($stackButton, 'o_hidden', "the stacked button should not have the .o_hidden class when we are in the line chart.");
+
+            await dom.click($stackButton);
+            assert.notOk(burndown_chart.renderer.props.stacked, "should be disabled and display a classic line chart.");
+
+            burndown_chart.destroy();
+        });
+
+        QUnit.test('check if the line chart is staked when the staked props is True', async function (assert) {
+            assert.expect(5);
+
+            const burndown_chart = await createView(this.burndown_chart);
+            assert.ok(burndown_chart.renderer.props.stacked, "should be stacked by default.");
+            assert.strictEqual(burndown_chart.renderer.props.mode, "line", "should be in line chart mode by default.");
```

Comment:
Should we retest this as it was tested in the previous test?

Reply:
I do this to be sure that stacked and mode are the same in each test, for instance, if someone changes the default view for the burndown chart, he will must check all tests and not only a test with these both lines. 
However, it is true the next assertions can be false if the mode is not line chart and stacked is not true. Thus, I'm agree with you, we can only do it in the first test. If you prefer to test it only once, I will change it.

Reply:
The idea behind my question is to try to be the most atomic we can in the tests. But is you think it is required, no problem for me 🙂

Reply:
Yeah, you are right, I changed it. 🙂